File: src/main/java/com/kmarzecki/communicator/model/CardEntity.java
package com.kmarzecki.communicator.model;

import lombok.*;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;
import java.util.List;
import java.util.UUID;

@Entity
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CardEntity {
    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(
            name = "UUID",
            strategy = "org.hibernate.id.UUIDGenerator"
    )
    @Column(name = "id", updatable = false, nullable = false)
    private UUID id;

    private String name;
    private String description;
    private Integer positionInLane;
    @ManyToOne
    private LaneEntity lane;
    @OneToMany
    private List<CardTaskEntity> tasks;
    @OneToMany
    private List<FileEntity> files;
}

File: src/main/kotlin/com/kmarzecki/trollo/model/CardEntity.kt
package com.kmarzecki.trollo.model

import lombok.*
import org.hibernate.annotations.GenericGenerator
import java.util.*
import javax.persistence.*

@Entity
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
data class CardEntity(
        @Id
        @GeneratedValue(generator = "UUID")
        @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
        @Column(name = "id", updatable = false, nullable = false)
        val id: UUID? = null,
        val name: String,
        val description: String,
        val positionInLane: Int,
        @ManyToOne
        val lane: LaneEntity,
        @OneToMany
        var tasks: MutableSet<CardTaskEntity>,
        @OneToMany
        val files: MutableSet<FileEntity>
)